enhancements/network/kubernetes-nmstate-operator-to-cvo.md

```diff
@@ -0,0 +1,284 @@
+- per node authority to write to the corresponding resource (think node restriction admission plugin in k8s as example)
+- any unique backup/restore semantics introduced by the component
+
+**A:** The resources composed of operator and handler (operand) resources.
+Check [this document](https://docs.google.com/document/d/1qFumb1P01NOIl1gCq79Q0I8BkeaA5UQY9spqlOJjy-I/edit?usp=sharing) for detailed description of resources.
+
+
+**Q:** What is Nmstate  future impact to other topologies (single node or externalized control planes)?
+For single node
+understanding what namespace hosts the component, and ensuring it fits in the cpuset isolation for mgmt components
+For externalized control planes
+is this needed only for the 'first' mgmt cluster and its workers?
+is this needed for every worker in all agnostic clusters, and therefore should be hosted with an externalized control plane because its needed to support any 'join worker' flow.
+
+**A:** SNO deployments using none platform will be unaffected as this operator will not be enabled by default.
+
+TODO - do we need a way to enable this for none-platform, particularly for UPI metal as ideally we’d want all metal customers to have access to the same day-2 API (even if not by default)?
+
+For externalized controlplane deployments this would be needed for both the management cluster and the externalized controlplane, so that the configuration of NICs can be handled in a consistent way - it’s an open question how we’d handle this in the hybrid platform case though...
+
+
+**Q:** is it ok to deliver the content in a release payload, but enable it via some other configuration knob?
+
+**A:** The desire is to enable a common enabled-by-default API for network configuration (initially only for selected platforms where that is commonly needed).
+
+If that approach is not acceptable then delivering via the release payload with some way to opt-in would be reasonable.
+
+
+**Q:** What is the knob to turn on/off the capability? This is related to the SLO chosen to deliver this new meta-operator and what configuration it reads to know to turn it on. For example, if this is for selected platforms, the component would read the Infrastructure CR to determine if it should enable it all, but is there another knob that would allow opt-in/out?
+
+**A:** The current proposal is to couple this to the platform detected in the Infrastructure CR, so it can be enabled by default on selected platforms (particularly baremetal).
+
+There could also be some other interface at the SLO level to override that behavior and/or to enable opt-in for platforms where it’s not enabled by default.
+
+
+### Goals
+
+- Provide NMState APIs by default on desired platforms, [NetworkManager configuration should be updated](https://github.com/openshift/enhancements/blob/master/enhancements/machine-config/mco-network-configuration.md#option-c-design) to support this capability on these platforms.
```

Comment:
After reading through this, it's not entirely clear what the "desired platforms" are. Is it just baremetal IPI?

Reply:
The initial target is just the baremetal platform, to support both CNV and general baremetal cases, but ref discussion above with @bengland2 there are other potential use-cases e.g OCS which wouldn't necessarily be specific to baremetal?

So I'd like to see this enabled by default for baremetal, with a way to opt-in for other platforms.

The other option that you mention above is to add the components to the release payload, but don't enable by default for any platform, then require some "advanced" customization step e.g manifest to enable (on any platform).

Reply:
For OpenShift Virtualization it would be important too to have a way to opt-in on other platforms.